src/overlays/actors/ovl_En_Ds2n/z_en_ds2n.c

```diff
@@ -28,24 +32,102 @@ const ActorInit En_Ds2n_InitVars = {
     (ActorFunc)EnDs2n_Draw,
 };
 
-#endif
+static AnimationInfo sAnimations[] = {
+    { &gDs2nIdleAnim, 1.0f, 0.0f, 0.0f, 0, 0.0f },
+};
+
+static Vec3f sZeroVec = { 0, 0, 0 };
+
+static TexturePtr sEyeTextures[] = { &gDs2nEyeOpenTex, &gDs2nEyeHalfTex, &gDs2nEyeClosedTex };
+
+void EnDs2n_SetupIdle(EnDs2n* this) {
+    this->blinkTimer = 20;
+    this->blinkState = 0;
+    Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimations, 0);
+    this->actionFunc = EnDs2n_Idle;
+}
+
+void EnDs2n_Idle(EnDs2n* this, GlobalContext* globalCtx) {
+    SubS_FillLimbRotTables(globalCtx, this->limbRotTableY, this->limbRotTableZ, DS2N_LIMB_MAX);
+}
+
+void EnDs2n_UpdateEyes(EnDs2n* this) {
+    s16 nextBlinkTime = this->blinkTimer - 1;
+
+    if (nextBlinkTime >= 3) {
+        this->blinkState = 0;
+        this->blinkTimer = nextBlinkTime;
+    } else if (nextBlinkTime == 0) {
+        this->blinkState = 2;
+        this->blinkTimer = (s32)(Rand_ZeroOne() * 60.0f) + 20;
+    } else {
+        this->blinkState = 1;
+        this->blinkTimer = nextBlinkTime;
+    }
+}
+
+void EnDs2n_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnDs2n* this = THIS;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 20.0f);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &gDs2nSkeleton, &gDs2nIdleAnim, NULL, NULL, 0);
+    EnDs2n_SetupIdle(this);
+}
+
+void EnDs2n_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnDs2n* this = THIS;
+
+    SkelAnime_Free(&this->skelAnime, globalCtx);
+}
+
+void EnDs2n_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnDs2n* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+    Actor_MoveWithGravity(&this->actor);
+    SkelAnime_Update(&this->skelAnime);
+
+    Actor_TrackPlayer(globalCtx, &this->actor, &this->headRot, &this->chestRot, this->actor.focus.pos);
+    EnDs2n_UpdateEyes(this);
+}
+
+s32 EnDs2n_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                            Actor* thisx) {
+    EnDs2n* this = THIS;
+
+    if (limbIndex == DS2N_LIMB_HEAD) {
+        Matrix_RotateXS(this->headRot.y, MTXMODE_APPLY);
+    }
 
-extern UNK_TYPE D_06008038;
+    return false;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ds2n/func_80AE1650.s")
+void EnDs2n_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    EnDs2n* this = THIS;
+    Vec3f focusOffset = sZeroVec;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ds2n/func_80AE16A0.s")
+    if ((limbIndex == DS2N_LIMB_01) || (limbIndex == DS2N_LIMB_0C) || (limbIndex == DS2N_LIMB_0F)) {
+        rot->y += (s32)(s16)Math_SinS(this->limbRotTableY[limbIndex]) * 0xC8;
+        rot->z += (s32)(s16)Math_CosS(this->limbRotTableZ[limbIndex]) * 0xC8;
```

Comment:
Is the second cast required?

Reply:
Yeah I remember this one specifically because m2c didnt give me the cast, and I had to wait for permuter to hint I needed one.